python/src/aac/plugins/validators/unique_names/_unique_names.py

```diff
@@ -0,0 +1,22 @@
+import logging
+
+from aac.lang.definitions.definition import Definition
+from aac.lang.language_context import LanguageContext
+from aac.plugins.validators._validator_result import ValidatorFindings, ValidatorResult
+
+
+PLUGIN_NAME = "Unique definition names"
+
+
+def validate_unique_names(
+    definition_under_test: Definition, target_schema_definition: Definition, language_context: LanguageContext, *validation_args
+) -> ValidatorResult:
+    """Search through the language context ensuring that all definition names are unique."""
+    findings = ValidatorFindings()
+
+    if definition_under_test.name in language_context.get_defined_types():
```

Comment:
This won't work for definitions already in the context, like validating the core spec. This should be changed to find all instances of definitions with the same name as the one under test, and if there is more than one definition with the name, then you can create a finding.